src/data/matrix/basic.lean

```diff
@@ -643,6 +670,27 @@ by simp [commute, semiconj_by]
 
 end comm_semiring
 
+section algebra
```

Comment:
`matrix/basic.lean` is quickly becoming a very big file, would it be possible to add this to a new file `data/matrix/algebra.lean` instead?

Reply:
I don't think that's a good idea - I shouldn't have to import extra files to get extra "basic" typeclasses on matrices.

At any rate, as a follow-up PR I plan to merge `matrix.scalar` with `algebra_map`, which should make up for anything added here.

Reply:
(as a justification: when we had the algebra in a different file, people didn't find it and wrote a duplicate instance!)

Reply:
If we were looking for a way to split this file up, one option I considered is to put all the multiplication stuff in its own file, and have `matrix/basic` just be (additive, transpose, reindexing, ...) operations on 2D arrays.

Reply:
I'm not so convinced that (mathematically) the best line to draw is one distinguishing addition from multiplication. But that's a conversation for another time.